packages/react-devtools-shared/src/backend/renderer.js

```diff
@@ -364,8 +364,19 @@ export function getInternalReactConstants(
       case IndeterminateComponent:
         return getDisplayName(resolvedType);
       case ForwardRef:
+        // maybe set `displayName` after `React.forwardRef()`
+        //
+        // ```
+        // const Component = React.forwardRef(WrapComponent);
+        // Component.displayName = 'OtherName'
```

Comment:
I don't know how this pattern came, I used this way to define the displayName for ref-forwarded component before too, and found it didn't work, and then I read the documentation carefully and can't found that kind of usage https://reactjs.org/docs/forwarding-refs.html#displaying-a-custom-name-in-devtools, so I changed my way to assign the displayName in my components

Reply:
Yes I agree we should align the logic with https://github.com/facebook/react/blob/master/packages/shared/getComponentName.js#L34, that will make life much easier, in the meantime do we need to update the doc for this case

Reply:
I know this way, it's so ugly.

And the reason why this docs exists is due to this bug.

Thinking what scene you need to use `React.forwardRef`, 
HOC, right?

So how can you set displayName after HOC returned in your code or babel?

Reply:
For HOC I updated as the documented way
![image](https://user-images.githubusercontent.com/2595058/70884608-6dfc3700-2011-11ea-8f4b-af455ac22923.png)

Reply:
Your code's result I have wrote [at the beginning of this PR](https://github.com/facebook/react/pull/17613#issue-353278549)

All of HOC returned components have the same name.

Example, some actual HOC code in [emotion](https://github.com/emotion-js/emotion/blob/master/packages/core/src/context.js#L20-L37)

and the used: 
- https://github.com/emotion-js/emotion/blob/master/packages/core/src/jsx.js#L98-L112
- https://github.com/emotion-js/emotion/blob/master/packages/styled-base/src/index.js#L81
- https://github.com/emotion-js/emotion/blob/master/packages/core/src/global.js#L24-L55